modules/qtx_module_loader.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+define( 'QTX_MODULE_STATE_UNDEFINED', 0 );
+define( 'QTX_MODULE_STATE_ACTIVE', 1 );
+define( 'QTX_MODULE_STATE_INACTIVE', 2 );
+define( 'QTX_MODULE_STATE_BLOCKED', 3 );
+
+require_once( QTRANSLATE_DIR . '/modules/qtx_module_setup.php' );
+
+/**
+ * Provide infos about the module states and the ability to load them. This never changes any state.
+ */
+class QTX_Module_Loader {
+    /**
+     * Get the modules previously activated in the options after validation for plugin integration on admin-side.
+     * Note these should be loaded before "qtranslate_init_language" is triggered.
+     *
+     * @return QTX_Module[]
+     * @see QTX_Admin_Modules::update_modules_state()
+     */
+    public static function get_active_modules() {
+        $options_modules = get_option( 'qtranslate_modules_state', array() );
+        if ( ! is_array( $options_modules ) ) {
+            return array();
+        }
+
+        $active_modules = array();
+        $modules        = QTX_Module_Setup::get_modules();
```

Comment:
Mmm... I just realized we don't even need this in the module loader!
All the module setup and code complexity should only be for the admin.